### DISCUSSION TITLE
Advantage over Microsoft Recommended CI/CD with Git Integration ?

### DISCUSSION THREAD

**Discussion by stas-sultanov** [Azure-Player/azure.datafactory.tools#317] [Show and tell]
Greetings!
Thank for your work! It is amazing!

I can clearly imagine how this tool is useful for an individual.
Just one engineer doing work with DataFactory.
All those granual deployments of each component.

What I can not understand is what advantages this tool provides to a Team of engineers doing their work with one DataFactory.
Advantages over the CI/CD approach with Git integration, which is provided by Microsoft. 
Here I refer to [new CI/CD flow](https://learn.microsoft.com/en-us/azure/data-factory/continuous-integration-delivery-improvements#the-new-cicd-flow)

All those granual deployments of the DataFactory components just won't work in the Team scenario.
It will require super extra effort to gather and compile changes made by several engineers and bring those changes to environment.
It will be much easier to just drop DataFactory and create new one and deploy complete state...

Could you please explain what I am missing...
Thank You!

**Comment by NowinskiK** [comment]
I have a whole session about this when explaining the differences, pros and cons:
:clapper: [Azure Data Factory - Deployment challenges](https://youtu.be/DG-TqdKlhhY)

**Comment by stas-sultanov** [comment]
@NowinskiK  I have watched a half of your session,
In the beginning you demonstrate old way to deploy ADF, which is described in MS documentation.

But my question is about the [NEW FLOW](https://learn.microsoft.com/en-us/azure/data-factory/continuous-integration-delivery-improvements#the-new-cicd-flow) not old one...

**Reply by NowinskiK** [reply]
My answer is referring to the new one either. 
So, new flow resolves some problems but not all of them, e.g. selective deployment, parametrisation.

**Reply by NowinskiK** [reply]
TBH, if you don't see the benefits of using deployment from code - you can stick to ARM Template (Microsoft) approach.
Even with that you can use #adftools to automate build (CI) pipeline with one step and other task to test deployed Linked Services.

**Comment by ken-vetter-absg** [comment]
I have many teams that DON'T want the global approach to deploying through the MS ARM template process.   
The benefit of this process is that many teams can work in the same ADF and deploy only wat they want to using the filter.md.
While the config.csv, covers parameters for the whole ADF, the filter.md allows a team to choose what goes into a pipeline and eventually on to Prod.  So, Developer 1 can deploy his Linked Service or Pipelines independently of other Developers work.  The ARM template process, grabs everything from the latest publishing which could force things to go to Prod before needed.
